### ISSUE TITLE
Start tag of empty `head` element is not omitted

### ISSUE THREAD

**Issue by 0f-0b** [syntax-tree/hast-util-to-html#43]
### Initial checklist

- [X] I read the [support docs](https://github.com/syntax-tree/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/syntax-tree/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/syntax-tree/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Asyntax-tree&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

hast-util-to-html@9.0.2

### Steps to reproduce

Run the code below.

```js
import { toHtml } from "hast-util-to-html";

console.log(toHtml(
  {
    type: "root",
    children: [
      {
        type: "element",
        tagName: "html",
        children: [
          {
            type: "element",
            tagName: "head",
            children: [],
          },
          {
            type: "element",
            tagName: "body",
            children: [],
          },
        ],
      },
    ],
  },
  { omitOptionalTags: true },
));
```

### Expected behavior

Empty output. The start tag of the `head` element is [omitted because the element is empty](https://html.spec.whatwg.org/multipage/syntax.html#optional-tags:the-head-element).

### Actual behavior

The output is `<head>`.

### Affected runtime and version

node@22.8.0

### Affected package manager and version

npm@10.8.3

**Comment by wooorm** [comment]
Can you please clarify actual/expected with code? Thanks

**Comment by 0f-0b** [comment]
> Can you please clarify actual/expected with code?

The actual output is a redundant start tag:

```text
<head>
```

The expected output is an empty string:

```text

```

**Comment by wooorm** [comment]
Did you see the end of the section you linked to?

> This section assumes that the document is conforming, in particular, that there are no [content model](https://html.spec.whatwg.org/multipage/dom.html#content-models) violations. Omitting tags in the fashion described in this section in a document that does not conform to the [content models](https://html.spec.whatwg.org/multipage/dom.html#content-models) described in this specification is likely to result in unexpected DOM differences (this is, in part, what the content models are designed to avoid).

Your empty document is not conforming: you need to have a `title`.

**Comment by 0f-0b** [comment]
> Your empty document is not conforming: you need to have a `title`.

It is, as an `iframe` `srcdoc` document.

**Comment by wooorm** [comment]
and is that what you’re doing? that seems unlikely 😅

**Comment by wooorm** [comment]
released in [9.0.3](https://github.com/syntax-tree/hast-util-to-html/releases/tag/9.0.3)!
